cli/kafka.go

```diff
@@ -57,11 +44,16 @@ func (cmd *Command) newKafkaRunner(cfgFile string) (*kafka.Runner, error) {
 		return nil, errors.Wrap(err, "cleaning config")
 	}
 
-	flds, err := kafka.ConfigToFields(cfg)
+	flds, err := kafka.ConfigToFields(cfg, idkCfg.PrimaryKeys)
 	if err != nil {
 		return nil, errors.Wrap(err, "getting fields from config")
 	}
 
+	// for avro, let the SchemaManager and IDK handle fields
+	if cfg.Encode == "avro" {
+		flds = nil
+	}
+
```

Comment:
Yep, make sense. Was trying a few different things so just placed it here to test and then never revisited. I I think I need to `return nil, nil` here because `*sqlBatcher.NewBatch()` is looking for `nil` vs and empty slice.
```
func (b *sqlBatcher) NewBatch(cfg fbbatch.Config, tbl *dax.Table, flds []*dax.Field) (fbbatch.RecordBatch, error) {
	fields := flds
	if b.fields != nil {
		fields = b.fields
	}
        ...
```